jaeger/cmd/check.go

```diff
@@ -93,6 +93,21 @@ func jaegerCategory(hc *healthcheck.HealthChecker) (*healthcheck.Category, error
 				return checkPodsRunning(podList.Items)
 			}))
 
+	checkers = append(checkers,
+		*healthcheck.NewChecker("jaeger injector pod is running").
+			WithHintAnchor("l5d-jaeger-jaeger-running").
+			Warning().
+			WithRetryDeadline(hc.RetryDeadline).
+			SurfaceErrorOnRetry().
+			WithCheck(func(ctx context.Context) error {
+				// Check for Jaeger Injector pod
+				podList, err := kubeAPI.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: "component=jaeger-injector"})
+				if err != nil {
+					return err
+				}
+				return checkPodsRunning(podList.Items)
```

Comment:
I think we also want to check that `len(podList.Items) > 0`.  If I scale the jaeger-injector deployment down to 0 replicas, this check passes.

Actually, it looks like this is an issue for the previous checks in this file as well.  Would you mind fixing those too while you're here?  Thanks!

Reply:
Sure!

Reply:
Just to confirm @adleong , if any of the pods (jaeger injector or collector or jaeger) are 0 in number, then, should the checks still succeed by just logging out a warning or should they fail?

Reply:
I think the checks should fail.

Reply:
Then, I guess for this, we can straightforwardly make all the pod related checks (like that of jaeger injector/jaeger/collector) to be of the type `Fatal` rather than `Warning`.

Reply:
Because, say, even after the `retryDeadline`, if  the pods are not in a running state or the pods are 0 in number, I feel that the `linkerd jaeger check` should fail. (currently, they succeed with a Warning)